src/budget_program.rs

```diff
@@ -21,24 +21,24 @@ pub enum BudgetError {
 }
 
 #[derive(Serialize, Deserialize, Debug, Clone, Default, PartialEq)]
-pub struct BudgetContract {
+pub struct BudgetProgram {
```

Comment:
This looks like the process state, not a program.

Reply:
Yea, this one is tough, because the code is implemented as methods on the state :).

Reply:
BudgetState?

Reply:
@garious BudgetState makes sense.  Does it make sense to put the entry points into a separate struct?  top level functions in the module (i kind of prefer that)